src/Api/PubnubApi/EventEngine/EffectDispatcher.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace PubnubApi.PubnubEventEngine
+{
+	public class EffectDispatcher
+	{
+		public enum DispatcherType
```

Comment:
```suggestion
		IEffectInvocationHandler currentEffectInvocationhandler;
```

I think disptacher should keep it at instance level. Otherwise there are chance when we receive next invocation, dispatcher can't find it as method execution have lost the handler reference.

Reply:
we don't need it. I removed it.